mozilla_django_oidc_db/views.py

```diff
@@ -325,10 +334,12 @@ def get_extra_params(self, request: HttpRequest) -> dict[str, str]:
         extra = super().get_extra_params(request)
         if kc_idp_hint := self.get_settings("OIDC_KEYCLOAK_IDP_HINT", ""):
             extra["kc_idp_hint"] = kc_idp_hint
+        if self.oidc_rp_scopes:
+            extra["scope"] = self.oidc_rp_scopes
```

Comment:
I'd move this above the `kc_idp_hint` because it's definitely far more standard and important than keycloak specific stuff, *and* also make it `extra.setdefault("scope", ...)` instead of overwriting what may have been set via `get_extra_params`.

I reckon for the Open Forms stuff we'll have to rely on more dynamic behaviour via `get_extra_params` to set the scope(s), so overriding this again if/when a default is set at the view level is not desired.

Reply:
I moved it above the `kc_idp_hint` but I am not sure I understand what you mean with:

> for the Open Forms stuff we'll have to rely on more dynamic behaviour via get_extra_params to set the scope(s)

since this is the `get_extra_params` mechanism to override the scopes specified in the configuration.

Reply:
for the Yivi/wallet stuff, we'll likely get extra parameters at runtime that we cannot statically define in code. so the class attribute will probably be some minimal fallback, and `get_extra_params` will contain a key `scope` from our custom plugin(s)

if the class attribute then undoes all that work, we don't have a way to implement our required behaviour.

Reply:
But I thought that in the OF auth plugin you do [this](https://github.com/open-formulieren/open-forms/blob/0ecd2c7b2a3750054390da7e4da13c77241cf3c2/src/openforms/authentication/contrib/digid_eherkenning_oidc/plugin.py#L96) (with the form data available):

```python
response = self.init_view(request, return_url=return_url)
```
And now you can do:
```python
digid_init = OIDCAuthenticationRequestInitView.as_view(
        identifier="digid-oidc", oidc_rp_scopes="<some scopes based on the form>"
    )
response = digid_init(request, return_url=return_url)
```
Like in this test: https://github.com/maykinmedia/mozilla-django-oidc-db/blob/1fdc9b500c47652f68110a31cb8f902451d4da7c/tests/test_init_flow.py#L80

Or am I still misunderstanding?

Reply:
That's possible but then we have to create the view dynamically every time instead of having it statically defined in the code 🤔 

I'm not very fond of having to add another "layer" like this, but I'm not sure if we can avoid it entirely either

Reply:
Maybe we can set the scopes on the request also? Like this for example:

```python
    oidc_init = OIDCAuthenticationRequestInitView.as_view(
        identifier="test-oidc", oidc_rp_scopes="email"
    )
    auth_request.oidc_rp_scopes = "not-email and-extra"
    redirect_response = oidc_init(auth_request)
```
Then we can both define it on the view, but we can still overwrite it with the request?

Reply:
I'm not a fan of inventing more magic attributes that are not statically defined on a request object, and especially attributes that sometimes do or don't exist. To stress that, static type checkers will complain about this too.

Finally, @robinmolen will have to chime in to confirm/deny, but right now we're assuming that the Yivi flow is based on passing a set of particular scopes, but who knows what else may be necessary to make it work, so I wouldn't restrict ourselves too much yet here.

Let's go with the dynamic view for now and then figure out what hurts in that setup and if we can live with it or not and whether there may be an elegant solution.